src/base/rss/rss_autodownloader.cpp

```diff
@@ -373,7 +373,7 @@ void AutoDownloader::addJobForArticle(Article *article)
 
 void AutoDownloader::processJob(const QSharedPointer<ProcessingJob> &job)
 {
-    for (AutoDownloadRule &rule : m_rules) {
+    for (AutoDownloadRule rule : qAsConst(m_rules)) {
```

Comment:
can it bee `const AutoDownloadRule &rule`?

Reply:
No... 3 lines below for const:
`error: 'this' argument to member function 'accepts' has type 'const RSS::AutoDownloadRule', but function is not marked const`
and 1 line below for reference:
`error: binding value of type 'const RSS::AutoDownloadRule' to reference to type 'RSS::AutoDownloadRule' drops 'const' qualifier`

I was wondering what's preferable here, keep the reference or use qAsConst?

Reply:
>I was wondering what's preferable here, keep the reference or use qAsConst?

I would avoid doing changes here, it is not legacy code.

Reply:
Actually this line is more than 1.5 years old, before qAsConst was introduced to qBt
https://github.com/qbittorrent/qBittorrent/commit/989a70fe60bb28c5c6bfdce86d6bec5949a833f0#diff-576c6aa10c20e1e0d491fd5cf5b81426R234
Maybe @glassez can say if it should be changed.

Reply:
This change is incorrect. Actual m_rules items should be modified inside this loop.